specs/agents/tracing-spans.md

```diff
@@ -54,7 +54,35 @@ The documentation should clarify that spans with `unknown` outcomes are ignored
 
 ### Span stack traces
 
-Spans may have an associated stack trace, in order to locate the associated source code that caused the span to occur. If there are many spans being collected this can cause a significant amount of overhead in the application, due to the capture, rendering, and transmission of potentially large stack traces. It is possible to limit the recording of span stack traces to only spans that are slower than a specified duration, using the config variable `ELASTIC_APM_SPAN_FRAMES_MIN_DURATION`.
+Spans may have an associated stack trace, in order to locate the associated
+source code that caused the span to occur. If there are many spans being
+collected this can cause a significant amount of overhead in the application,
+due to the capture, rendering, and transmission of potentially large stack
+traces. It is possible to limit the recording of span stack traces to only
+spans that are slower than a specified duration, using the config variable
+`ELASTIC_APM_SPAN_STACK_TRACE_MIN_DURATION`. (Previously
+`ELASTIC_APM_SPAN_FRAMES_MIN_DURATION`.)
+
+#### `span_stack_trace_min_duration` configuration
+
+Sets the minimum duration of a span for which stack frames/traces will be
+captured.
+
+This option is case-insensitive.
+
+|                |   |
+|----------------|---|
+| Valid options  | `-1`, `0`, [duration](configuration.md#configuration-value-types) with default suffix `ms` |
+| Default        | `5ms` (`5`)  |
+| Dynamic        | `true` |
+| Central config | `true` |
```

Comment:
Would we add both or just one of `span_stack_trace_min_duration` and `span_frames_min_duration` to central config? If we remove the legacy option, the UI element will be gone but the config may still be stored in the index. Also, if you're using a new Kibana and old agents, you'd think that you can configure the threshold but it'd get silently ignored by agents.

Reply:
We could potentially have a period where we support both. This will likely not be the last config option we rename, so we should probably have a policy for the switchover period. I don't know what's best.

(As an aside, the idea of upgrading my elastic stack but pinning my agent versions makes me shudder.)

Reply:
We discussed this in the agents weekly meeting. The “best” solution is likely to replace the old name with the new name in the UI, and have that UI write to *both* the old and new config option. This solution has code smell (hard-coded handling of the semantic differences for example) but would be good from a user perspective.

Reply:
Note that none of the UI folks were on that call. ;)
The current agent config code in APM UI (https://github.com/elastic/kibana/tree/master/x-pack/plugins/apm/common/agent_configuration/ and https://github.com/elastic/kibana/tree/master/x-pack/plugins/apm/public/components/app/Settings/AgentConfigurations/) doesn't look to have anything like this handling right now, so might be a pain to implement.